research/m-out-of-n.md

```diff
@@ -0,0 +1,43 @@
+Let say we have input data. And we need to create `n` data blocks that:
+- having any `m` of blocks, input data can be recovered,
+- having less than `m` of blocks, input data cannot be recovered.
+
+Resulting data blocks are understood as user secrets and input data is intended for a private key recovery.
+
+# Algorithm 1.
+
+This is simpler one but provides larger user secrets in terms of bytes length.
+
+1. Divide input data into ${n\choose n-(m-1)}$ data chunks of equal length.
+2. Include every chunk in exactly `n-(m-1)` blocks, the way that any two different chunks are not included in the same `n-(m-1)` blocks.
+
+Proof of correctness. 
+There are ${n\choose n-(m-1)}$ chunks and ${n\choose n-(m-1)}$ different sets of `n-(m-1)` blocks.
+Thus the algorithm is doable.
+Assume you know at least `m` blocks and consider a given chunk. 
+The chunk is included in `n-(m-1)` blocks. It holds `m+n-(m-1)>n`.
+Thus the chunk is included in at least one block you know and in consequence all input data can be recovered.
+Assume you know less than `m` blocks. 
+There exists `n-(m-1)` blocks that you do not know. 
+There exists a chunk that is included only in these `n-(m-1)` blocks.
+Thus there exists a chunk you do not know and input data cannot be recovered.
+
+Note that 32 bytes chunks do not weaken a private key. It means that, if an adversary knows `n-1` chunks, the private key is still secure.
```

Comment:
We have ${n\choose n-(m-1)}$ data chunks, so it should say "if an adversary knows ${n\choose n-(m-1)}$ - 1 chunks"